solidity/dashboard/src/components/Header.jsx

```diff
@@ -1,30 +1,70 @@
-import React, { useContext } from "react"
-import * as Icons from "./Icons"
-import { Web3Context } from "./WithWeb3Context"
-import { Web3Status } from "./Web3Status"
-import { MenuButton } from "./MenuButton"
-import AddressShortcut from "./AddressShortcut"
-import { NetworkStatus } from "./NetworkStatus"
+import React, { useState, useCallback } from "react"
+import { NavLink } from "react-router-dom"
+import { isEmptyArray } from "../utils/array.utils"
+import { useContext } from "react"
 
-const Header = (props) => {
-  const { yourAddress } = useContext(Web3Context)
+const Header = () => {
+  const { title, subLinks } = useHeaderContext()
```

Comment:
So, I _think_ if the page structure looks more like what I described above, we can have `PageWrapper` be in charge of rendering the header, at which point the title and sublinks can propagate hierarchically instead of out-of-band via context. I'm not 100% sure of this, but from what I see it should be true.

Reply:
I thought about it but opted for a context. The main idea with `HeaderContext` is that we only have the `Header` component in one place and for me, this solution fits better to the struct of app:
```
nav header
nav content
```
But now I think rendering a `Header` in a `PageWrapper` makes a lot of sense and it is less "complicated".